pytorch_lightning/trainer/distrib_data_parallel.py

```diff
@@ -535,7 +552,17 @@ def ddp_train(self, process_idx, model, is_master=False, proc_offset=0):
         model = model.configure_ddp(model, device_ids)
 
         # continue training routine
-        self.run_pretrain_routine(model)
+        results = self.run_pretrain_routine(model)
+
+        # clean up memory
+        torch.cuda.empty_cache()
+
+        if self.global_rank == 0 and q is not None:
+            q.put(self.checkpoint_callback.best_model_path)
```

Comment:
this feels hacky, what are we trying to do here? return the state of a callback to the main node? why put this specific attribute in the queue?

Reply:
I think he did this so that one can call .test in the main process, which will access the best model to test on. But I agree with you, this seems fragile and dangerous to modify state across processes, there's gotta be a different way. Could one put the whole trainer in the queue?

Reply:
i tried that but it didn’t work. i think in a different PR we can do something like state_dict for the trainer.

Reply:
i agree this isn't optimal but let's get a release that fixes all the test issues (which this PR does) and then we can figure out a longer term strategy